File: learning/runtime/objectKeyValues/objectKeyValues/ViewController.m
//
//  ViewController.m
//  objectKeyValues
//
//  Created by WZH on 16/8/12.
//  Copyright © 2016年 王振海. All rights reserved.
//

#import "ViewController.h"
#import "NSObject+keyValues.h"
#import "student.h"
#import "School.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    School *school = [[School alloc] init];
    school.age = 100;
    school.name = @"要塞";
    student *st = [[student alloc] init];
//    st.age = 10;
    st.name = @"wzh";
    school.st = st;
    
    NSLog(@"%@",[school modelToKeyValues]);
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Demo/Swift/20.Swift3.0 - 构造函数KVC/20.Swift3.0 - 构造函数KVC/ViewController.swift
//
//  ViewController.swift
//  20.Swift3.0 - 构造函数KVC
//
//  Created by WZH on 16/10/16.
//  Copyright © 2016年 limpid. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        let p = Person(dict: ["name":"Hayder", "age": 109, "title":"BOSS"])
        
        print("\(p.name) \(p.age)")
        
        let s = Student(dict: ["name":"Hayder", "age": 109, "title":"BOSS","no":"1234"])
        
        print("\(s.name) \(s.no)")

    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

